### PR TITLE
Conditionally refetch positions more frequently

### PR DESCRIPTION
**Pull request by everdimension**
Set refetch interval to 4 seconds within 2 minutes of a last known submited transaction

### CODE REVIEW

File: src/ui/transactions/usePositionsRefetchInterval.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+/** A flag for perf optimisation: we don't need to track "useNow()" after a tx has been handled */
+let handledTimestamp: number | null = null;
+
+/**
+ * This hook returns a small refetchInterval value
+ * within two minutes of a last known submitted transaction
+ * {false} is a valid value for react query which means no refetch interval
+ */
+export function usePositionsRefetchInterval(fallback: number | false) {
```
Comment (zerts) [inline src/ui/transactions/usePositionsRefetchInterval.ts:47]: Let's make a named param. To make it `usePositionsRefetchInterval({fallbackInterval: false})`

File: src/ui/transactions/usePositionsRefetchInterval.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+const URGENT_REFETCH_INTERVAl = 4000; // 4 seconds
+
+function useNow(enabled: boolean) {
+  const [now, setNow] = useState(() => Date.now());
+  useEffect(() => {
+    if (enabled) {
+      // doesn't really need to match URGENT_REFETCH_INTERVAl,
+      // but also we don't need to refresh more frequently than that
+      const refreshInterval = URGENT_REFETCH_INTERVAl;
+      const id = setInterval(() => {
```
Comment (zerts) [inline src/ui/transactions/usePositionsRefetchInterval.ts:30]: Is it ok that we will update `now` only 4 sec after `enabled` was set to `true`?

**Review by zerts — APPROVED** [review]

File: src/modules/zerion-api/hooks/useWalletPositions.ts [hunk 3]
```diff
@@ -25,7 +25,7 @@ export function useHttpAddressPositions(
     suspense?: boolean;
     enabled?: boolean;
     keepPreviousData?: boolean;
-    refetchInterval?: number;
+    refetchInterval?: number | false;
```
[the remark this answers is not in the document]
Comment (everdimension) [reply src/modules/zerion-api/hooks/useWalletPositions.ts:28]: 🤔 I don't think it should be

File: src/ui/transactions/usePositionsRefetchInterval.ts (hunk 2 above)
Comment (everdimension) [reply src/ui/transactions/usePositionsRefetchInterval.ts:30]: I think so

**Pull request merged**